### PR TITLE
Fix for replicant name change bug + fixed tests + added eslint plugin for hooks

### PR DESCRIPTION
**Pull request by CarlosFdez**
The package.json probably needs to run jest as well, however prettier stops it from running in its entirety. I don't know how to properly handle that scenario so I'll leave it for the next in line.

Let me know if there's anything you need changed.

### CODE REVIEW

File: src/__tests__/use-replicant.tsx [hunk 1]
```diff
@@ -18,44 +30,119 @@ class Replicant extends EventEmitter {
...
+	const replicant = allReplicants.get('counter');
+
+	// We know its not undefined cause we asserted it, but we have a picky linter.
+	if (typeof replicant === 'undefined') return;
+	if (renderResult.container.firstChild === null) return;
+
+	const initialValue = replicant.value as number;
+	expect(replicant.value).toBe(0);
+
+	fireEvent.click(renderResult.container.firstChild as Element);
```
Comment (CarlosFdez) [inline src/__tests__/use-replicant.tsx]: Pre-emptively gonna state, despite fireEvent.click(container.firstChild) is the recommended way of doing it in the tutorial, the type system really didn't like it. Therefore I casted it. If you know of an alternative way of doing this then it'd certainly help a lot.

One alternative might be using the react testing library specifically for hooks. Not sure about doing that though.

**Comment by Hoishin** [comment]
Can you add `jest` in `"test"` script in package.json?

**Comment by CarlosFdez** [comment]
I could, but it won't run cause both prettier and lint stops all further execution. I also don't know what to call the npm command that'll actually run jest (since run-s only runs npm commands), as "test" is already taken. Could use a suggestion.

I just fixed lint locally (eslintignore both esm/ and cjs/ works), but prettier is still broken and I have zero clue what to do with it.

**Comment by Hoishin** [comment]
I added jest in test script in master branch, you can merge or rebase to have that.

For prettier and eslint, they are passing on Travis CI so it should be fine

**Comment by CarlosFdez** [comment]
If Prettier is passing on travis, that reveals that the problem is most likely a line endings issue because I'm on Windows. I ran format-fix on my machine and git said that there were no differences, however prettier started passing. I think you need to make it ignore CRLF vs LF somehow.

Anyways, done. Using && to run jest separately from run-s was the right call and I'm a bit salty I didn't think of that.

**Comment by CarlosFdez** [comment]
Let me know if you still need something from this PR.

**Comment by Hoishin** [comment]
Sorry, I'm going to look into it once again this week.

**Comment by Hoishin** [comment]
Looks good. Thank you so much!

**Pull request merged**